### ISSUE TITLE
Add evaluation metrics and Categorical Cross-Entropy loss function

### ISSUE THREAD

**Issue by thiago-buarqque** [thiago-buarqque/NeuraRust#7]
Add evaluation metrics:

1. Accuracy
2. Precision
3. Recall
4. F1-Score

Add Categorical Cross-Entropy loss function
